app/services/atmosphere/cloud/vm_creator.rb

```diff
@@ -39,6 +51,30 @@ def execute
 
     attr_reader :user_data
 
+    def set_azure_endpoints(params)
+      tcp_endpoints_str = ''
+      udp_endpoints_str = ''
+      @appliance_type.port_mapping_templates.
+        each do |pmt|
+          if pmt.transport_protocol == :tcp
+            # 22 ssh is present by default on Azure for linux
+            # and if it is specified explicitly it causes error
+            # same for rdp 3389
+            unless pmt.target_port == 22 || pmt.target_port == 3389
+              tcp_endpoints_str << "#{pmt.target_port}:#{pmt.target_port},"
+            end
+          else
+            udp_endpoints_str << "#{pmt.target_port}:#{pmt.target_port},"
+          end
+        end
+      if tcp_endpoints_str.present?
+        params[:tcp_endpoints] = tcp_endpoints_str.chomp!(',')
+      end
+      if udp_endpoints_str.present?
+        params[:udp_endpoints] = udp_endpoints_str.chomp!(',')
+      end
+    end
```

Comment:
I understand that only predefined PMT are supported. There will not be possible to add new redirection once appliance is started.